[export] Replicate exportdb examples and tests in oss.

zhxchen17 opened the pull request:
Summary: Initial work to copy source to OSS for exportdb and make sure tests can run properly.

Test Plan: test_export

Differential Revision: D46369152



cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy

angelayi commented on test/export/test_db.py:
```diff
@@ -0,0 +1,114 @@
+        model = case.model
+        with torchdynamo.config.patch(asdict(_DynamoConfig())):
+            inputs = normalize_inputs(case.example_inputs)
+            exported_model, _ = torchdynamo.export(
```
should this be torch.export eventually?

zhxchen17 replied:
yeah, I'm looking at @tugsbayasgalan

angelayi commented on torch/_export/db/gen_example.py:
```diff
@@ -0,0 +1,28 @@
+import os
```
what is this file for again?

zhxchen17 replied:
this is a script to generate a new example file with empty code.

angelayi reviewed (approved):
I think the tests are failing bc you need an empty __init__.py file in the export/db directory but everything else looks fine to me!

zhxchen17 commented:
@pytorchbot merge

zhxchen17 committed:
[export] Replicate exportdb examples and tests in oss. (#102769)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/102769

Initial work to copy source to OSS for exportdb and make sure tests can run properly.

Test Plan: test_export

Reviewed By: angelayi

Differential Revision: D46369152

fbshipit-source-id: 202b90e6b1aa29183f78d20ef47f0be1d0810892

zhxchen17 commented:
@pytorchbot merge
